tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Krivodonova.cpp

```diff
@@ -0,0 +1,2900 @@
+  // the math easier but still test thoroughly.
+  Limiter krivodonova{std::vector<double>(
+      Spectral::maximum_number_of_points<Spectral::Basis::Legendre>, 0.99)};
+
+  NeighborData<dim, typename Limiter::PackagedData> neighbor_data{};
+
+  const Element<dim> element(ElementId<dim>{0}, {});
+  // We don't care about the ElementId for these tests, just the direction.
+  Limiter::PackagedData& package_data_up_xi = neighbor_data[std::make_pair(
+      Direction<dim>::upper_xi(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_lo_xi = neighbor_data[std::make_pair(
+      Direction<dim>::lower_xi(), ElementId<dim>{0})];
+
+  package_data_up_xi.modal_volume_data.initialize(num_pts);
+  package_data_up_xi.mesh = mesh;
+  package_data_lo_xi.modal_volume_data.initialize(num_pts);
+  package_data_lo_xi.mesh = mesh;
+
+  Scalar<DataVector> nodal_scalar_data_to_limit(num_pts, 0.0);
+  tnsr::I<DataVector, dim> nodal_vector_data_to_limit(num_pts, 0.0);
+  Scalar<DataVector> nodal_scalar_expected(num_pts, 0.0);
+  tnsr::I<DataVector, dim> nodal_vector_expected(num_pts, 0.0);
+
+  // Limit top coeff from upper neighbor
+  to_nodal_coefficients(&get(nodal_scalar_data_to_limit), {3.0, -2.0, 2.0, 1.0},
+                        mesh);
+  get(get<::Tags::Modal<ScalarTag<0>>>(package_data_up_xi.modal_volume_data)) =
+      ModalVector{7.0, 3.0, 2.5, 0.0};
+  get(get<::Tags::Modal<ScalarTag<0>>>(package_data_lo_xi.modal_volume_data)) =
+      ModalVector{1.0, -5.0, 1.0e-18, 0.0};
+  to_nodal_coefficients(&get(nodal_scalar_expected),
+                        {3.0, -2.0, 2.0, 0.5 * 0.99}, mesh);
+
+  // Limit top upper neighbor, limit second lower neighbor
+  to_nodal_coefficients(&nodal_vector_data_to_limit.get(0),
+                        {3.0, 2.0, 2.0, 1.0}, mesh);
+  get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_up_xi.modal_volume_data)
+      .get(0) = ModalVector{7.0, 5.0, 2.5, 0.0};
+  get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_lo_xi.modal_volume_data)
+      .get(0) = ModalVector{-2.1, 1.1, 1.0e-18, 0.0};
+  to_nodal_coefficients(&nodal_vector_expected.get(0),
+                        {3.0, 2.0, 0.9 * 0.99, 0.5 * 0.99}, mesh);
+
+  krivodonova(&nodal_scalar_data_to_limit, &nodal_vector_data_to_limit, element,
+              mesh, neighbor_data);
+  CHECK_ITERABLE_APPROX(get(nodal_scalar_data_to_limit),
+                        get(nodal_scalar_expected));
+  for (size_t i = 0; i < dim; ++i) {
+    CAPTURE(i);
+    CHECK_ITERABLE_APPROX(nodal_vector_data_to_limit.get(i),
+                          nodal_vector_expected.get(i));
+  }
+}
+
+void run() noexcept {
+  INFO("Testing 1d limiter");
+  for (size_t order = Spectral::minimum_number_of_points<
+           Spectral::Basis::Legendre, Spectral::Quadrature::GaussLobatto>;
+       order < Spectral::maximum_number_of_points<Spectral::Basis::Legendre>;
+       ++order) {
+    package_data(order);
+  }
+  two_neighbors();
+  test_different_values_different_tensors();
```

Comment:
For increased clarity, I suggest renaming these functions to
- `test_package_data`, to avoid any ambiguity with the function being tested
- `test_limiting_two_neighbors`
- `test_limiting_different_values_different_tensors`

Could also do similar renames in 2D,3D cases.

Reply:
Sure, but just to confirm, the function names will then be `test_1d::test_package_data`, etc. This is what you want?

Reply:
Yes, I agree it's slightly redundant, but it will make it clearer at the call site (or invoke as `test_1d::package_data`, to the same effect).